one-ui/src/main/java/com/marklogic/hub/oneui/listener/UIDeployListener.java

```diff
@@ -0,0 +1,71 @@
+package com.marklogic.hub.oneui.listener;
+
+import com.marklogic.hub.deploy.util.HubDeployStatusListener;
+import com.marklogic.hub.oneui.exceptions.ProjectDirectoryException;
+import com.marklogic.hub.oneui.models.StatusMessage;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.messaging.simp.SimpMessagingTemplate;
+
+public class UIDeployListener implements HubDeployStatusListener {
+    protected final Logger logger = LoggerFactory.getLogger(UIDeployListener.class);
+
+    static final Map<String, Object> stompHeaders = new HashMap<>() {{
+        put("content-type","application/json");
+    }};
+
+    private SimpMessagingTemplate template;
+
+    private boolean isUninstall;
+
+    private int lastPercentageComplete;
+
+    private Exception exception;
+
+    public UIDeployListener(SimpMessagingTemplate template, boolean isUninstall) {
+        this.template = template;
+        this.isUninstall = isUninstall;
+    }
+
+    @Override
+    public void onStatusChange(int percentComplete, String message) {
+        if (percentComplete >= 0) {
+            logger.info(percentComplete + "% " + message);
+            lastPercentageComplete = percentComplete;
+            String msg = parseMessage(message);
+            template.convertAndSend("/topic/install-status", new StatusMessage(percentComplete, msg), stompHeaders);
+        }
+    }
+
+    @Override
+    public void onError(String commandName, Exception exception) {
+        String message = "Error encountered running command: " + commandName;
+        String msg = parseMessage(message);
+        template.convertAndSend("/topic/install-status", new StatusMessage(lastPercentageComplete, msg));
+        logger.error(message, exception);
+        this.exception = exception;
+        if (exception instanceof IOException || exception.getCause() instanceof IOException) {
+            this.exception = new ProjectDirectoryException("Your user account does not have write permissions to the installation directory.",
```

Comment:
This seems like another error that the user can't possibly resolve. It seems like a confusing error message too, as the user will think it's their MarkLogic user account. But it's the OS user account that matters here - i.e. which OS user was used to fire up the middle tier. 

I think a more appropriate error message is (same with above) - "Installation (or uninstallation) failed, please contact a system administrator; cause: " + e.getMessage(). That makes it clear that the expected user can't do anything to resolve the issue, and an IT admin will need to get involved.

Reply:
@rjrudin do we want to have the message for suggesion or message?

Reply:
I don't know how the "suggestion" is used, but I suppose the message could be "Installation failed; please contact a system administrator for assistance." and the suggestion would be "Installation error: " + e.getMessage().